.github/workflows/cd.yml

```diff
@@ -0,0 +1,89 @@
+name: CD
+on: 
+  workflow_dispatch:
+
+jobs:
+  terraform:
+    name: 'Setup Azure Infrastructure'
+    runs-on: ubuntu-latest
+    env:
+      ARM_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
+      ARM_CLIENT_SECRET: ${{ secrets.AZURE_CLIENT_SECRET }}
+      ARM_SUBSCRIPTION_ID: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+      ARM_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
+
+    defaults:
+      run:
+        working-directory: ./terraform
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Ensure directory permissions
+        run: sudo -s
+        
+      - name: Azure Login
+        run: az login --service-principal -u $ARM_CLIENT_ID -p $ARM_CLIENT_SECRET --tenant $ARM_TENANT_ID
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v1
+        with:
+          # This wrapper must be disabled in order to correctly capture output from terraform CLI into a bash var
+          # See https://github.com/hashicorp/setup-terraform/issues/20
+          terraform_wrapper: false
+
+      - name: Terraform Init
+        run: terraform init
+

```

Comment:
Should we add a `terraform plan` step in between to validate the state and allow us to fail early?

Reply:
Done! thanks